Rebase onto keplergl/kepler.gl

james-willis opened the pull request:
I'm wondering if the change to the reactIntl here: bindings/kepler.gl-jupyter/js/webpack/build-html.js

Will resolve this bug: https://www.notion.so/wherobots/Exported-Kepler-Visualizations-Crash-when-Pulling-Out-Menu-c472ef79e4d64faab4f9e7c90a97ef2a?pvs=6&utm_content=c472ef79-e4d6-4faa-b4f9-e7c90a97ef2a&utm_campaign=T046Z383PCL&n=slack&n=slack_link_unfurl

furqaankhan reviewed (approved):
LGTM! @daicjohn thoughts?

igorDykhta committed:
[feat] Call get after inject to create full cache (#2647)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Disable polygon filter menu for non-polygon features (#2652)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Making sure animated spinner has border width CSS prop set (#2651)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] Exporting missing types for PlaybackControls (#2650)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Map controls tooltips break drag event positioning (#2649)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Reset default values when DropdownList component unmounts (#2648)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] add autoFocus prop in TypeHead (#2646)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] Add className for LayerManager (#2629)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Updated type data-utils getColumnFormatter method (#2640)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Upgrade react-router from 3.2.5 to 3.2.6 (#2637)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] prevent second shadow effect (#2631)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[fix] Improved map bounds calculation and handled latitude issues (#2632)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] Move create or update filter action (#2636)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Fix] legend wasn't interactive in shadow DOM (#2630)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[chore] Update @loaders to 4.1.1 (#2638)

* update loaders to 4.1.1

mars-dlx committed:
[fix] Fixed disappearing animation time control (#2625)

* Fixed disapiaring animation time control

Signed-off-by: Sergey Markov <mar.ser.dlx@gmail.com>

igorDykhta committed:
[feat] Layer Column Mode (#2662)

- Layer Column Mode
-  add ColumnMode support for TripLayer 
- Choose between geoJson and CSV columns for trip layer
- create new component for column mode config
- calculate animation config when update layer column
- Fill all columns from column mode when initializing it from prop
- Auto detect lat, lng, altitude field pairs
- Auto create trip layer if a dataset contain field pairs, timestamp, and `id` | `uuid` fie
- Add column config component test

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] Split out column mode config into separate component (#2663)

- This PR splits out a more general-purpose ColumnModeConfig that we can use separately outside of the layer panel.
- Looks the same as before, changes here should not affect user experience

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Feat] support create geojson path from point csv in polygon layer (#2664)

- Support `columnMode: table` in Polygon layer
- Improve layer column auto-assign logic
- Add modal info for trip and geojson layer

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Feat] Add neighbor column mode to arc layer, support arc from hex (#2665)

- Cleanup columnMode logic, remove use of `assignColumnsByColumnMode`
- Add neighbor column mode to arc layer
- Support arc from hex_id
- [fix] Ensure Polygon layer filtering in table columns mode works
- [fix] Timestamp in the Trips layer is not required after switching from Polygon

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Feat] add geojson column mode for point layer (#2666)

* add point column mode in point layer
* update point layer geojson column mode; update test cases;

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

heshan0131 committed:
[Chore] Adding application config (#2658)

* [unfork] Adding application config (proposal for discussion) (#862)

* Application config

* Update kepler-gl.tsx

* Trying to fix failing tests

* Move applicationConfig to separate file to avoid test errors

* Revert "Trying to fix failing tests"

This reverts commit 1ee283aaa9b24fe968928ceb673402c35a022a71.

* Moving export to the end

* Adding missing dependency

* Comment and more explicit export

* Try to fix import errors

* Remove unnecessary change

* Moved applicationConfig to utils in another attempt to avoid errors in CI

* Reorg

* update demo-app yarn.ock

Signed-off-by: Shan He <heshan0131@gmail.com>

---------

Signed-off-by: Shan He <heshan0131@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

heshan0131 committed:
[unfork] [Enhancement] call layer methods to validate visconfig when switching dataset (#2676)

* [Enhancement] call layer methods to validate visconfig when switching dataset

---------
Signed-off-by: Shan He <heshan0131@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

heshan0131 committed:
[Feat] Support radius legend  (#2677)

* Show radius legend in point layer

Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

ilyabo committed:
[Fix] Tooltip not working in exported HTML map (#2556)

* Update deps in html export

Signed-off-by: Ilya Boyandin <ilya@boyandin.me>

* Adding yarn version to package.json

Signed-off-by: Ilya Boyandin <ilya@boyandin.me>

---------

Signed-off-by: Ilya Boyandin <ilya@boyandin.me>

heshan0131 committed:
[Chore] Remove default props and react-onclickoutside in react functional components (#2679)

Use of defaultProps and findDomNode has been deprecated. This change tries to remove use of both, to avoid React warnings in the dev console.

- Remove defaultProps in functional components, some class components are still using defaultProps
- Add default props to functional components method to be used
- Fix replace-component example
- Remove use of react-clickoutside (which uses findDomNode)

There are still warnings about deprecated APIs from the following dependencies, further dependency upgrade is needed

- react-modal uses findDomNode
- react-markdown uses defaultProps
- react-router: uses childContextTypes

heshan0131 committed:
[Feat] Dynamic map lib config (#2678)

* Support dynamic map lib using the application config

---------

Signed-off-by: Shan He <heshan0131@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

heshan0131 committed:
[Enhancement] Add billboard and fadeTrail toggles (#2684)

- Added billboard toggle for point and icon layers
- Added billboard and fadeTrail toggles for trips layer

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Igor Dykhta <igorDykhta@users.noreply.github.com>

igorDykhta committed:
[feat] Edit color legend value (#2681)

- feat: Allow color legend category edit
- Able to reset previously customized value
- Updated factory type definitions
- Add disableEdit to mapControl.mapLegend

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[feat] Implemented ability to invert time series trend colors (#2692)

- Pass through properties to invert time series trend colors

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] Time filter syncing (#2690)

* Time filter syncing
* Splitting setFilterUpdater into multiple functions to reduce complexity as required by eslint
* mergeDomain when adding synced filter
* Histogram plot supports multi-histogram
* Adding colorsByDataId

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] add deck.gl onFilteredItemsChange callback to DeckGl overlays (#2691)

* add layerFilteredItemsChange to visState
* add onFilteredItemsChange callback to deck.gl layers
* save filteredItemCount to layer
* add getFilterItemCount to layer

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Feat] Replaced filter enlarged with view: side | enlarged | minified - part 2 (#2537)

- Replaced filter enlarged with view: onToggleMinify

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

heshan0131 committed:
add map control buttons back (#2709)

- [Bug] add map control buttons back 

Signed-off-by: Shan He <heshan0131@gmail.com>

TangSY committed:
fix: remove duplicate "https:" in example (#2711)

igorDykhta committed:
[chore] Add runGpuFilterForPlot to export, ts changes to KeplerTableModel  (#2702)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Prevent bottom time widget crash (#895) (#2703)

- Prevent bottom time widget crash

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] Updated time filter sync style (#2705)

* [feat] Updated time filter sync style

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Bug] Fix synced time filter loaded value not saved (#892) (#2706)

- Move adjustValueToFilterDomain out, only call when add datasets are validated

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Shan He <heshan0131@gmail.com>

igorDykhta committed:
[Feat] add color picker to single color selector (#2699)

* [Feat] add color picker to single color selector

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[BUG] fix gpu filter update trigger attribute update in every render (#2707)

* [BUG] fix gpu filter update trigger attribute update in every render

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
Allow function return type of getData in getFilterValueAccessor (#2708)

- changes to getFilterValueAccessor: Allow function return type from getData in getFilterValueAccessor

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[faat] deckgl-arrow-layers module (#2680)

* deckgl-arrow-layers-module

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] Layer animation (#2713)

* enable timesteps for setLayerAnimationTimeUpdater
* do not reset animationConfig when animatable
* [Bug] call setLayerAnimationTime after currentTime changed in updateAnimationDomain 
* [fix]: Fixed animation config domain set to null when toggling layer visibility 
* Fix: Changes to splitMaps layer visibility were overwritten by when layer animation enabled

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[fix] filter fields based on timestamp (#2714)

- Syncable fields should only be timestamp
- Calculate bar width prevent .x0 of undefined crash

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Enhancement] Synced filter small tuneup to synced filter panel (#2715)

- small tuneup to synced filter panel
- Added regression test for the applyFilterFieldName fix
- Remove unnecessary import

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] Sync filter with layer timeline (#2718)

- Able to display timeline on filter plot

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Time Sync bugfixes (#2721)

- hide layer timelines if a filter is synced and not enlarged
- Add guard to prevent crash in binByTime
- fix 0 display in synced dataset filter panel

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] Improve timeline sync filer UI (#2722)

- Improve time filter and sync timeline UI

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Giuseppe Macrì <macri.giuseppe@gmail.com>

MilheiroSantos committed:
[Doc] Improve keplergl-jupyter documentation (#2697)

- Removed reference to deprecated {} icon that no longer exists in current version.
- Added FAQ & Troubleshoot entry to table-of-contents
- Clarified sentence on Windows support

Signed-off-by: Pedro Santos <milheiro.santos@gmail.com>

igorDykhta committed:
[Improvement] Improved radius legend number formatting (#2726)

- pass fieldType to formatNumber

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] time sync bugfixes (#2723)

- [fix] Export syncTimelineMode correctly to be able to reload it
- [feat] show sync timeline icon
- [fix] Hide layer sync icon when filter is not synced 

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Giuseppe Macrì <macri.giuseppe@gmail.com>

igorDykhta committed:
[Bug] fix yaxis chat doesn't update (#2724)

- fix yaxis chat doesn't update
- Sync time filter with layer animation 

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Fixed synced filter domain and interval calculation (#2725)

- Fixed synced filter domain and interval calculation

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

indranildeveloper committed:
[chore] Update Code examples in API Reference Get Started page (#2727)

Signed-off-by: Indranil Halder <indranilhalder.dev@gmail.com>

igorDykhta committed:
[feat] experimental support for ARROW:extension:point; support for arrrow:wkb for geojson layer (#2716)

- adds support for ARROW:extension:name EXTENSION_NAME.POINT position columns in Point, Line, Arc, Text layers.
- can pick geoarrow point column as source for Heatmap layer.
- interleaved xy geoarrow point vector is passed directly as a position data attribute for rendering.
- separate lon / lat columns are converted to geoarrow point vectors (only when source file is an .arrow file).
- this PR doesn't affect existing loading and support of non-geo columns.
- support for arrow:wkb in geojson layer. Atm features are transformed from wkb to features to binary features as getBinaryGeometriesFromArrow doesn't support arrow:wkb.

Known issues:
- text layer supplied with ARROW:extension:point column can only render string arrow columns (no values offsets).

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

indranildeveloper committed:
[chore] Docs action page import updates (#2729)

* Update imports in api reference actions page and root readme
* Reverted the markdown href changes

Signed-off-by: Indranil Halder <indranilhalder.dev@gmail.com>

indranildeveloper committed:
Custom Initial State and Forward Actions Docs update (#2731)

* Custom Initial State and Forward Actions Docs update
* Update Kepler GL component import

Signed-off-by: Indranil Halder <indranilhalder.dev@gmail.com>

igorDykhta committed:
[fix] restore arc and line layers in non-geoarrow modes (#2732)

* [fix] pass line and arc layer missing props

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] Support custom breaks in color scale (#2739)

-Add SCALE_TYPE.custom option
-Save color breaks in colorRange.colorMap
-add customBreaks: true to layer.colorUI.colorRange.colorRangeConfig to render color breaks panel
-Make color scale selector a seperate component
-Create factory for color-selector, dimension-scale-selector, color-range-selector, custom-palette

Signed-off-by: Shan He <heshan0131@gmail.com>
Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Custom color scale fixes (#2770)

- [Bug] Fix custom color scale removing break error
- [Bug] Fix customPalette undefined
- [Bug] fix color legend intl id not defined

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[fix] Time Sync fixes and tests (#2771)

- [fix] Fixed time range filter panel dataset field overflow
- [test] Added more time filter/layer tests for time filter animation update action
- [fix] Loading synced filter value correctly

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

lixun910 committed:
[Feat] Kepler.gl AI Assistant [1] (#2735)

igorDykhta committed:
[chore] common-utils module (#2773)

* [chore] common-utils module

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

indranildeveloper committed:
[chore] Updated imports for Kepler GL Reducers in docs (#2774)

Signed-off-by: Indranil Halder <indranilhalder.dev@gmail.com>

indranildeveloper committed:
[fix] Update the path to relative path in utils (#2775)

Signed-off-by: Indranil Halder <indranilhalder.dev@gmail.com>

indranildeveloper committed:
Update the keplergl processors update (#2776)

Signed-off-by: Indranil Halder <indranilhalder.dev@gmail.com>

igorDykhta committed:
[feat] create new dataset action (#2778)

* [feat] create new dataset action

- createNewDataEntry now returns a react-palm task to create or update a dataset asynchronously.
- updateVisDataUpdater now returns tasks to create or update a dataset asynchronously, and once done triggers createNewDatasetSuccess action.
- refactoring of demo-app App and Container to functional components

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Shan He <heshan0131@gmail.com>

igorDykhta committed:
[chore] 3.1.0-alpha.0 release (#2789)

* 3.1.0-alpha.0 changes

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] fix for package publishing (#2790)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] fix npmpublish action (#2791)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] npmpublish fixes - set npm registry manually (#2792)

* npmpublish fixes - set npm registry manually

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] prepublishOnly command fixes (#2793)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fix examples - proper publicPath to include bundle chunks (#2795)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Fix for Try Sample Data (#2796)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] align upload icon (#2799)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fixes for exported maps (#2807)

- fix broken maplibre gl css link
- fix umd build
- use react 18.3.1 in exported maps
- temp fix for missing datasets in exported maps

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Foursquare storage provider improvements (#2800)

- Change default height of the auth popup to show "Continue with Google" and bottom links.
- Show an info helper that not all FSQ maps are expected to show up in Load From Storage modal.
- fix constant reloading of map from link mode
- fix jest tests

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

indranildeveloper committed:
[chore] Local Development Guide Update (#2806)

- Updated the local development guide. Also, added the steps to set up Kepler.GL locally on Windows.

Signed-off-by: Indranil Halder <indranilhalder.dev@gmail.com>

Ritesh9876 committed:
[Bug] Fix issue with React StrictMode causing Dataset table to not display (#2683)

Signed-off-by: Ritesh9876 <riteshmangdare@gmail.com>

lixun910 committed:
[Feat] AI Assistant [2] (#2777)

igorDykhta committed:
[fix] disable Share Map for FSQ provider (#2808)

- disable FSQ storage provider in Share Map modal as the provider doesn't support public raw kepler maps atm.

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Updated plot when changing cross filters (#2801)

* [fix] Updated plot when changing cross filters

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] timeline refactoring (#2802)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] show sync layer animations when there is one dataset (#2803)

- show sync layer animations when there is one dataset
- Fix: TypeError: e.valueAccessor is not a function

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Fix] Don't modify height for with fixed height enabled; Height UI unification (#2804)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

1Maxnet1 committed:
[chore] Fixes in README.md (#2810)

- correct alt tag
- update links from http to https

dependabot[bot] committed:
Bump cross-spawn from 6.0.5 to 6.0.6 (#2772)

Bumps [cross-spawn](https://github.com/moxystudio/node-cross-spawn) from 6.0.5 to 6.0.6.
- [Changelog](https://github.com/moxystudio/node-cross-spawn/blob/v6.0.6/CHANGELOG.md)
- [Commits](https://github.com/moxystudio/node-cross-spawn/compare/v6.0.5...v6.0.6)

---
updated-dependencies:
- dependency-name: cross-spawn
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump elliptic from 6.5.7 to 6.6.0 (#2720)

Bumps [elliptic](https://github.com/indutny/elliptic) from 6.5.7 to 6.6.0.
- [Commits](https://github.com/indutny/elliptic/compare/v6.5.7...v6.6.0)

---
updated-dependencies:
- dependency-name: elliptic
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump cross-spawn from 6.0.5 to 6.0.6 in /website (#2817)

Bumps [cross-spawn](https://github.com/moxystudio/node-cross-spawn) from 6.0.5 to 6.0.6.
- [Changelog](https://github.com/moxystudio/node-cross-spawn/blob/v6.0.6/CHANGELOG.md)
- [Commits](https://github.com/moxystudio/node-cross-spawn/compare/v6.0.5...v6.0.6)

---
updated-dependencies:
- dependency-name: cross-spawn
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump elliptic from 6.5.7 to 6.6.1 in /bindings/kepler.gl-jupyter/js (#2818)

Bumps [elliptic](https://github.com/indutny/elliptic) from 6.5.7 to 6.6.1.
- [Commits](https://github.com/indutny/elliptic/compare/v6.5.7...v6.6.1)

---
updated-dependencies:
- dependency-name: elliptic
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

igorDykhta committed:
[feat] Show selected fields in the tooltip for aggregation layers (#2814)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Typescript 4.4 fixes (#2816)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Line layer is not displayed for between hex ids (#2820)

- fix Line layer is not displayed for between hex ID's
- rename type ColumnModeConfig since ColumnModeConfig
- export ColumnModeConfig

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[FEAT] support domain.domainStops in layer color, render color legend based on zoom (#2815)

* use layer.getColorScale in color legend
* add zoom to attribute accessors
* layer color legend by zoom and improve legend number format

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[fix] Fix multiple field filtering in setFilterUpdater (#2821)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Feat] add h3 typed column (#2822)

Detect H3 column as a new h3-type column, like the existing geocolumn.

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] adjust time range filter on value set (#2823)

- additional type updates for filter types where we get rid of BaseFilter properties that we override in all the filter types

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Expand legend (#2824)

* Expand legend
* Flex alignment
* Only show collapse button when enableColorBy is true
* Not subtracting size of 4 map control buttons when isExport
* Using customizable actionIcons

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Chore] Minor ts refactoring (#2825)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Feat] Add display format setting for table/tooltip (#2826)

- [Feat] Add display format setting for table/tooltip
- ts changes

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] ts fixes (#2827)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] getSelectedFeature (#2832)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] Apply new legend style (#2831)

- pass through `className` to legend
- css updates to reflect new designs

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Prevent duplicate legend in line and arc layers (#2830)

Before the fix we were showing two identical color legends for each line and arc layer (when selecting "color based on field").
This PR changes the following:
- If color based on field, only show one color legend.
- If single color is selected, continue to show both rectangle color swatch boxes, but put text next to each one indicating “source” and “target” instead of empty space.

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Feat] handle layer color scale by field.domainQuantiles (#2829)

* handle layer domain quantile
* add threshold scale

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[Feat] Redesign color range to use chormajs and d3 color function (#2835)

- Replace default color ranges with color palette object, with .colors(step), .linear function to auto generate colors / ramps
- Add more d3 color chroma palette
- Add more color blind safe palette
- Save and load color range from old config 

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

sahitono committed:
[Enhancement] support mapbox url protocol (#2693)

Signed-off-by: Sahit Tuntas Sadono <sahitsadono@gmail.com>

dependabot[bot] committed:
Bump nanoid from 3.3.7 to 3.3.8 (#2837)

Bumps [nanoid](https://github.com/ai/nanoid) from 3.3.7 to 3.3.8.
- [Release notes](https://github.com/ai/nanoid/releases)
- [Changelog](https://github.com/ai/nanoid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ai/nanoid/compare/3.3.7...3.3.8)

---
updated-dependencies:
- dependency-name: nanoid
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump nanoid from 3.3.7 to 3.3.8 in /website (#2840)

Bumps [nanoid](https://github.com/ai/nanoid) from 3.3.7 to 3.3.8.
- [Release notes](https://github.com/ai/nanoid/releases)
- [Changelog](https://github.com/ai/nanoid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ai/nanoid/compare/3.3.7...3.3.8)

---
updated-dependencies:
- dependency-name: nanoid
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

eltociear committed:
[docs] update demo-app/README.md (#2847)

typo enviroment -> environment

igorDykhta committed:
[fix] fix yarn cover regression (#2846)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] vector tile layer (#2839)

- This PR adds Vector Tile layer to Kepler with support initial support of mapbox vector tiles and remote .pmtiles (with mimeType application/vnd.mapbox-vector-tile)

Ritesh9876 committed:
[docs] Replace yarn global add with yarn dlx for puppeteer installation (#2849)

Signed-off-by: Ritesh9876 <riteshmangdare@gmail.com>

igorDykhta committed:
[fix] custom palette issues (#2850)

* 1 update fix of custom palette issues 2 restrict custom steps to 20, hide + button if needed 3 update scale function to handle some special cases: not enough k groups
* add name|type|category to updateCustomPalette if customBreaks
* add test cases for redesigned color range feature
* update test cases to increase coverage

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[feat] color scale histogram (#2851)

- color scale histogram

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[feat] histogram and point layer fixes (#2852)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fix for custom color scales with tile layer (#2853)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Fixed time filter bug where points located at the borders of the domains were not correctly displayed (#2854)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] 3.1.0-alpha.2 release (#2855)

- bump version for release
- temporarily disable vector tile layer plumbing
- ts fixes required for npm-publish
- prepublish to prepublishOnly script

igorDykhta committed:
[chore] fix missing dependencies in workspaces (#2856)

- fix missing dependencies in workspaces
- fix circular dependencies
- fix types
- fixes for release

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] 3.1.0-alpha.3 release (#2857)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

chrisgervang committed:
[chore] Create SECURITY.md (#2845)

igorDykhta committed:
[fix] change process to globalThis.process (#2865)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] add fsq color palettes (#2862)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Feat] replaceDataInMap action - save colorsByDataId (#2859)

- changes to support replaceDataInMap action in synced filters
- tests for replaceDataInMapUpdater with Synced syncedTimeFilter

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Feat] Add custom color scale for aggregate layers (#2860)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[chore] ts refactoring (#2861)

- move several base layer types to layer.d.ts
- other ts changes

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Remove legend layer groups height capping (#2864)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

lixun910 committed:
[Release] Candidate Kepler-Jupyter v0.3.4 (#2588)

igorDykhta committed:
[feat] add LayerToggleVisibility for single splitMap (#2863)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

lixun910 committed:
[Fix] add_data() function in keplergl-jupyter (#2867)

Publish Keplergl-jupyter 0.3.5

igorDykhta committed:
[feat] Free positioning of the legend (#2874)

- Legend can be moved around
- Position is saved as offsets to the closest corner to handle window resizes

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[feat] Support custom ordinal color scale on string field in layer config (#2868)

- Support custom ordinal color scale on string field in layer config
- histogram from ordinal
- pass className to StyledLegend
- fix for color palette colorWidths undefined

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] Consolidate vis state tests (#2869)

- fix for jest test ran into an issue where warnings were showing up about logging after a test was finished

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] pass strings to color palette inputs (#2873)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] show scale options for Point Count in aggregation layers (#2870)

- show scale options for Point Count in aggregation layers

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Fix] incorrect multi-dataset filter domain (#2871)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] applyFilterConfig action (#2872)

- Introduces a new action `applyFilterConfig`
- Changes `FilterPanelHeader` so that it accepts an array of actions in order to pass custom actions.

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[Chore] Add composedReducerSchema to schema manager (#2633)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] Add several vis state mergers combineConfigs and improve TS (#2634)

- Adds `combineConfigs` to some vis state saved configs that are non-array complex objects:
  1. `interactionConfig`
  2. `layerBlending`
  3. `overlayBlending`
  4. `editor`
  
  Each of these has inline comments (and unit tests) explaining the decisions made on how to combine values in different situations. For example, `interactionConfig.brush.size` gets aggregated into keeping the single largest value among all brush sizes.

- Improves and fixes some TS defs.

- Adds jest unit tests for newly added `combineConfigs`.

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <iboyandin@foursquare.com>

igorDykhta committed:
[fix] Custom Color Scale fixes (#2875)

- [Chore] add exports for scenegraph to layers/index
- [fix] Apply custom column format on color legend
- [feat] support custom color scale for layers using colorScale.byZoom
- [fix] disable custom ordinal scale
- [chore] add FSQ Cool Tone color palette

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] check for allBins (#2876)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

isaacbrodsky committed:
[fix] copy map config style (#2878)

Signed-off-by: Isaac Brodsky <isaac@isaacbrodsky.com>

Ritesh9876 committed:
[Bug] Prevent dragging legend component outside of container (#2885)

Signed-off-by: Ritesh9876 <riteshmangdare@gmail.com>

igorDykhta committed:
[Feat] Custom color scale for categorical/ordinal field (#2880)

- [feat] Implement custom color scale for categorical field
- [fix] ColorMap is not cleared when field or scale change
- [fix] Fix crash when custom ordinal scale does not have colorMap
- [fix] fix custom stroke color initiation

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Use floating-ui to deal with closing on click outside (#2886)

- use `floating-ui` in `AnimationSpeedSlider` and `ColorSelector` to deal with closing on click outside.
- For both of the above components, this fixes the component from disappearing immediately in a programable environment even when clicking internally, bypassing important onClick event logic.

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Fixed issue when mapstate latitude or langitude are out of bounds (#2882)

- fixed issue when mapstate latitude or langitude are out of bounds

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] export TimeFieldFilterProps (#2883)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] Adding setAnimationConfig action (#2887)

- Added setAnimationConfig action
- Passing filter to PlaybackControls

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Tippy tooltips in animation controls (#2888)

- Replacing react-tooltip tooltips in animation controls which don't work correctly when used in programable environments

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Handle selecting geojson polygon with missing properties (#2889)

- fix crashing geojson polygon isClosed check for the editor selected feature when the feature `properties` are nullish
- don't set the selected feature in map popover if it is effectively empty

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Add ellipsis in LayerTypeListItem (#2890)

- Add ellipsis in LayerTypeListItem
- bump dndkit
- bump caniuse-lite

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fixed onOptionSelected handler when option is null (#2891)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Prevent infinite useEffects loop in range-plot (#2892)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] Upgrade styled components to v6 (#2894)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Image export legend positioning (#2895)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

Ritesh9876 committed:
[Chore] Fix lint error for Register actions (#2896)

Signed-off-by: Ritesh9876 <riteshmangdare@gmail.com>

igorDykhta committed:
[feat] mapbox and maplibre simultaneous support (#2897)

- switch between basemap style libraries
- add back mapbox styles in Basemap Style tab

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

lixun910 committed:
[Fix] add_data() function failed in keplergl-jupyter (#2901)

igorDykhta committed:
[feat] Vector Tile layer fixes (#2899)

- show Vector Tile layer tab
- changes to radius controls
- fix highlighted filled polygons
- hide show data table icon
- display loading spinner while we are loading metadata
- display errors metadata loading failed
- fix for custom ordinal crash when all colors are deleted
- collect dataset attributions from metadata and show in the lower right corner 
- refetch tile dataset metadata createNewDataEntry

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] 3.1.0-alpha.4 (#2866)

- bump version
- ts fixes for npm publish
- missing license headers

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fix website commit section (#2908)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] adjust tooltip style (#2910)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fix styled-components version in exported map (#2907)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Don't show 0 when description is empty (#2913)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] opening multiple examples breaks url (#2915)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] check for geoarrow extension in geojson layer (#2916)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] retry 3.1.0-alpha.4 release (#2917)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fix exported html map (#2918)

- import missing css that isn't part of the main bundle anymore.
- replace withState custom state (demo - undefined) with lenses.
- alpha 5 version

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] vector tile layer fixes (#2911)

- dataset type change: vectorTile to vector-tile
- vector tile dataset metadata: type to remoteTileFormat
- support for remote examples

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fix csv table examples for layers (#2921)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fix neighbor mode hovering crash in arc/line layers (#2920)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fixes for style editor preview and error (#2919)

- fix style preview 
- take into account basemap library switching in add-map-style-modal
- trigger new Map components with url, token or library change in order to clear cache of loaded styles. Otherwise 'style.load' events aren't triggered when input changes from correct > wrong > correct config.

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] support for table plugin in demo examples and privately stored datasets (#2923)

- support for custom table plugin + demo examples
- support for custom table plugin + privately stored datasets
- isAppleDevice util

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fix demo-app yarn start (#2924)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] demo-app resolve to proper node_modules (#2925)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] duckdb module placeholder (#2926)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] duckdb module updates (#2927)

- UI for SQL Query plugin
- DuckDb logic to ingest and generate datasets

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] add dot.env, updates to demo-app build, update gitignore (#2928)

- add dot.env plugin for demo-app
- updates to demo-app build
- update gitignore

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] use saved map config for saved maps instead of zoom in to data (#2929)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] remove dependencies from useEffect (componentDidMount) (#2930)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

lixun910 committed:
[Chore] esbuild website to speedup netlify build (#2904)

igorDykhta committed:
[chore] 3.1.0-alpha.6 (#2931)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] don't auto create point layer from vector tiles (#2932)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] minZoom and maxZoom for examples (#2933)

- KeplerGlSchema drops minZoom and maxZoom for mapState.
- for examples inject custom minZoom and maxZoom from remote configs.

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

ilyabo committed:
Fixes for using in a vite app (#2898)

- Fixing window import which breaks in a vite app
- Exporting useDimensions with a fix

Signed-off-by: Ilya Boyandin <ilya@boyandin.me>
Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ihor Dykhta <dikhta.igor@gmail.com>

mars-dlx committed:
[fix] Added 24 limit for maximum zoom (#2635)

Signed-off-by: Sergey Markov <mar.ser.dlx@gmail.com>
Co-authored-by: Sergey Markov <mar.ser.dlx@gmail.com>

lixun910 committed:
[Feat] AI Assistant Query (#2819)

lixun910 committed:
[Misc] Update demo-app README.md (#2934)

lixun910 committed:
[Chore] Remove SQL plugin for AI Assistant (#2938)

igorDykhta committed:
[feat] loading indicator (#2936)

- setLoadingIndicator action
- show Loading... text when async dataset loading is in progress

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

ilyabo committed:
[feat] duckdb plugin (#2798)

- optional duckdb plugin support in demo-app

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] hotfix for arrow saving / loading (#2943)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] 3.1.0-alpha.7 (#2937)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] demo-app bump kepler.gl version (#2944)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] plumbing for DuckDB plugin support (#2949)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

lixun910 committed:
[Chore] Add DeepSeek in Ai Assistant (#2946)

distefanodevinj committed:
testing mp4 embed with gitbook (#2951)

Signed-off-by: distefanodevinj <ddistefano@foursquare.com>

igorDykhta committed:
[fix] adjust margin for map save modal to show Save button without scrolling (#2954)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] arrow tables - save timestamps as iso date string (#2953)

- convert numerical timestamps from arrow tables into iso date strings before saving.

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] banner with extra release info (#2955)

- banner with extra info for 3.1 release
- once minified, the user won't see the banner again during next sessions
- can be disabled via application config

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

distefanodevinj committed:
[docs] update docs for Kepler.gl release 3.1 (#2941)

- Added a new release notes/what's new section to show new user-facing features (not a replacement for the changelog, which is a complete timeline of all kepler.gl development)
- Vector tile layer page
- AI assistant page
- SQL explorer/duckdb page
- Updates to ToCs 

---------

Signed-off-by: distefanodevinj <ddistefano@foursquare.com>
Co-authored-by: Igor Dykhta <igorDykhta@users.noreply.github.com>
Co-authored-by: Xun Li <lixun910@gmail.com>
Co-authored-by: Ilya Boyandin <ilya@boyandin.me>

igorDykhta committed:
[fix] DuckDB: make query result title more reusable (#2956)

- Change default query result table from `Query Result (1)` to `query_result_1`

---------

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] vector tiles refactoring (#2945)

- enable vector tile layer
- enable release banner


---------

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] kepler.gl 3.1.0 (#2947)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] update banner mesasge (#2957)



Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] update demo-app version (#2958)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fix for minzoom in examples (#2959)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] FSQ Storage provider temp fix (#2960)

- FSQ Storage provider temp fix for
`Error while fetching maps: Cannot read properties of undefined (reading 'map')`

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fix geojson and trip layer crash without data (#2964)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

rbavery committed:
[Jupyter][Fix] convert datetimes to str so they can convert to json (#2968)

lixun910 committed:
[keplergl-jupyter] Release v0.3.7 #2969

ironicbrew committed:
[chore] Rename ".env.template "  to ".env.template" to prevent git clone fail (#2976)

igorDykhta committed:
[fix] fix for point column suggestion not working (#2974)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] FSQ storage provider - use prompt instead of auto login after logout (#2975)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] check for required env variables in demo-app and output a warning (#2977)

- kepler demo app will crash with cryptic `process isn't defined` when not all required env variables are set
- check for required env variables in demo-app and output a warning to the console

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] heatmap renders nothing with black color or duplicate color (#2978)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

fkauker-fsq committed:
Update README.md (#2981)

Looks like the vis academy link is dead moving to the repo

yharby committed:
[fix] AI Assistant sends messages to 127.0.0.1 instead of remote Ollama URL (keplergl#2984) (#2985)

chrisgervang committed:
chore: update to hubble.gl 1.4 (#2987)

Signed-off-by: Chris Gervang <chris@gervang.com>

igorDykhta committed:
[fix] restore suport for string wkb; save binary wkb as hex wkb (#2982)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] export geoarrow to CSV as geojson (#2988)

- [x] geoarrow.polygon save / load as stringified geojson
- [x] geoarrow.linestring save / load as stringified geojson
- [x] geoarrow.point save / load as stringified geojson
- [x] hex wkb string save / load as stringified geojson

---------

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: Ilya Boyandin <ilya@boyandin.me>

ironicbrew committed:
Improved Props and Gettings Started Docs (#2993)

* Rename ".env.template "  to ".env.template"

* Update README.md

* Added working template to get-started.md

igorDykhta committed:
[chore] updates to website (#2992)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] arrow text labels from non-string source vectors (#2990)

- add missing support for apache-arrow point columns and text labels from numeric fields (calculate offsets).
- geoarrow-text-layer requires Utf8 vector.
- in case of non Utf8 vector create a temporary Utf8 vector with the same layout of batches as in the source vector.

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] improvements for layer type change logic (#2995)

- when switching between Geojson > Point > Heatmap layers the columns end up in an invalid state when using geoarrow point fileds which are also recognized as geojson data.

- add an optional `verifyField` function to verify whether a column is valid for a column mode.
- a layer can end up with unsupported `columnMode` after layer type is switched, so add extra check and try to use one of the default layer configurations in case of unsupported `columnMode`.
- `findDefaultLayerProps` also returns `altProps` with layer configurations that shouldn't be created by default, but are still valid choices.
- fix for Geojson layer not reacting to column when in `COLUMN_MODE_TABLE` mode.

---------

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[feat] improvements to duckDB column type handling (#2970)

This PR intends to preserve column types between different types of ingestion into Kepler and DuckDb

- timestamps stored as strings from Arrow tables are recognized as timestamps. 
- apply extra metadata from table.schema.metadata (geoparquet files). 
- DuckDB geometry is automatically casted to WKB, and properly marked with geoarrow extensions.
- DuckDB column types and query result Arrow table types consolidation.
- Apply extra logic only to the last select query.
- geoarrow constants to constants module
- add getSampleForTypeAnalyzeArrow to support and not fail for arrow data
- arrowSchemaToFields accepts extra info from DuckDB table schemas. JSON type gets GEOMETRY_FROM_STRING type, GEOMETRY with geoarrow metadata gets GEOMETRY type, timestamp ...
- fix in validateInputData - check analyzerType only for current field
- fix in validateInputData - support arrow input data

---------

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Transform binary buffers to hex wkb when saved to json/hmtl maps (#2998)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] Fix lint issues displayed on GitHub's File Changes page (#3001)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

dependabot[bot] committed:
Bump elliptic from 6.6.0 to 6.6.1 (#2997)

Bumps [elliptic](https://github.com/indutny/elliptic) from 6.6.0 to 6.6.1.
- [Commits](https://github.com/indutny/elliptic/compare/v6.6.0...v6.6.1)

---
updated-dependencies:
- dependency-name: elliptic
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump esbuild from 0.23.1 to 0.25.0 in /examples/demo-app (#2994)

Bumps [esbuild](https://github.com/evanw/esbuild) from 0.23.1 to 0.25.0.
- [Release notes](https://github.com/evanw/esbuild/releases)
- [Changelog](https://github.com/evanw/esbuild/blob/main/CHANGELOG-2024.md)
- [Commits](https://github.com/evanw/esbuild/compare/v0.23.1...v0.25.0)

---
updated-dependencies:
- dependency-name: esbuild
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump lodash from 4.17.19 to 4.17.21 in /src/deckgl-layers (#2858)

* Bump lodash from 4.17.19 to 4.17.21 in /src/deckgl-layers

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.19 to 4.17.21.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.19...4.17.21)

---
updated-dependencies:
- dependency-name: lodash
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* bump

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ihor Dykhta <dikhta.igor@gmail.com>

dependabot[bot] committed:
Bump fast-xml-parser from 4.4.0 to 4.5.0 (#2688)

* Bump fast-xml-parser from 4.4.0 to 4.5.0

Bumps [fast-xml-parser](https://github.com/NaturalIntelligence/fast-xml-parser) from 4.4.0 to 4.5.0.
- [Release notes](https://github.com/NaturalIntelligence/fast-xml-parser/releases)
- [Changelog](https://github.com/NaturalIntelligence/fast-xml-parser/blob/master/CHANGELOG.md)
- [Commits](https://github.com/NaturalIntelligence/fast-xml-parser/compare/v4.4.0...v4.5.0)

---
updated-dependencies:
- dependency-name: fast-xml-parser
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* bump

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

* bump

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ihor Dykhta <dikhta.igor@gmail.com>

dependabot[bot] committed:
Bump nanoid from 3.3.7 to 3.3.8 in /bindings/kepler.gl-jupyter/js (#2906)

Bumps [nanoid](https://github.com/ai/nanoid) from 3.3.7 to 3.3.8.
- [Release notes](https://github.com/ai/nanoid/releases)
- [Changelog](https://github.com/ai/nanoid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ai/nanoid/compare/3.3.7...3.3.8)

---
updated-dependencies:
- dependency-name: nanoid
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump express from 4.19.2 to 4.21.0 (#2655)

Bumps [express](https://github.com/expressjs/express) from 4.19.2 to 4.21.0.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/4.21.0/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.19.2...4.21.0)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

badbye committed:
[fix] Update geojson-utils.ts to support GeometryCollections (#2059)

Signed-off-by: Yalei Du <yaleidu93@gmail.com>

jwhitaker-swiftnav committed:
[Bug] make sure the RangeBrush updates on slider range changes (#2047)

Signed-off-by: Jarrad Whitaker <jwhitaker-swiftnav@users.noreply.github.com>

igorDykhta committed:
[fix] Fix for More than one copy of react-palm was loaded error message (#3007)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fix for Icon layer UI (#3003)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] adjust getZoomFactor for icon layer (#3004)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fix for geocoder coordinates (#3002)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] Fix Save map action for FSQ provider (overwrite logic) (#3006)

- This PR fixes the map save functionality for the Foursquare provider when overwriting an existing map. 
- update uploadMap to properly fetch and update map data during an overwrite.
- add a new savedMapId state field in the provider state updaters


Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

---------

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

mars-dlx committed:
[chore] changes to webpack.config path separators (#2623)

Signed-off-by: Sergey Markov <mar.ser.dlx@gmail.com>
Co-authored-by: Igor Dykhta <igorDykhta@users.noreply.github.com>

igorDykhta committed:
[feat] DuckDb plugin: drag and drop file directly as table (#2952)

- DuckDB plugin: drag and drop file directly as table

---------

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] update umd example to latest stable release (#3010)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] DuckDB: update schema after running a query (#3009)

* [fix] update schema after a query

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

* ts

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

* nit

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

---------

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[chore] fix react deprecation warnings (#3011)

- update react-markdown
- disable react-markdown in tape tests (EMS).
- updates to load-remote-map and errors from FormattedMessage
- FileDrop to functional component
- GeocoderPanel to functional component
- remove deprecated call to findDOMNode (left after previous refactoring?)
- data preview table - warning for missing style for react-virtualized Grid

---------

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] schema panel displays temp table (#3014)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

igorDykhta committed:
[fix] fixes for vector-tile layer (#3013)

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>
